python/lsst/meas/extensions/piff/piffPsfDeterminer.py

```diff
@@ -72,10 +72,27 @@ def _validateGalsimInterpolant(name: str) -> bool:
 
 
 class PiffPsfDeterminerConfig(BasePsfDeterminerTask.ConfigClass):
+    spatialOrderPerBand = pexConfig.DictField(
+        doc="Per-band spatial order for PSF kernel creation. "
+        "Ignored if piffPsfConfigYaml is set.",
+        keytype=str,
+        itemtype=int,
+        default={
+            "u": 2,
+            "g": 4,
+            "r": 4,
+            "i": 4,
+            "z": 4,
+            "y": 4,
+            "fallback": 2,
+        },
+    )
     spatialOrder = pexConfig.Field[int](
         doc="Spatial order for PSF kernel creation. "
         "Ignored if piffPsfConfigYaml is set.",
-        default=2,
+        optional=True,
+        deprecated="This field is no longer used (it has been replaced with the "
+        "spatialOrderPerBand dict config). Will be removed after v30.",
```

Comment:
Fly by review. Was there an RFC to deprecate it? For now, it might make sense to use this to set the fallback on (but eventually deprecate it). For instance, `spatialOrder = self.config.spatialOrderPerBand.get(band, self.config.spatialOrder)

Reply:
Oh, no...this requires an RFC?  @TallJimbo, do you have any advice on how to best proceed with this?

Reply:
I'm afraid so. I don't know if there are exceptions for 1-day-before-the-deadline though!
https://developer.lsst.io/stack/deprecating-interfaces.html#deprecation-procedure

Reply:
In any case, while the parent class `BasePsfDeterminer` doesn't guarantee the presence of `spatialOrder`, many other PSF determiner classes do (`PSFEx`, `PCA` etc.) and it would be good to keep them all as homogenous as possible. So we should address this after next week.

Reply:
Ok, thanks a lot Arun!  I am editing along the lines of what you suggest above.  I'm leaving the OG spatialOrder config and adding the spatialOrderPerBand dict config but setting the dict as empty by default (so will "fallback" to spatialOrder in that case) and adding the override in `obs_lsst`.  Am I right in thinking this procedure does not require RFC (and we can clean all of this up post DP1 deadlines)?

Reply:
Add the new option now, RFC deprecating the old one later.  Ideally you'd fall back to using the old option somehow; maybe if there's no configuration for the band you've got?

Reply:
I _think_ that agrees with what I've got now.  Please have a look!

Reply:
Yup, looks good!